src/Form/UpdateMediaItemForAccount.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\media_mpx\Form;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\media\Entity\Media;
+use Drupal\media_mpx\Service\UpdateVideoItem\UpdateVideoItem;
+use Drupal\media_mpx\Service\UpdateVideoItem\UpdateVideoItemRequest;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Form class to import / update a single mpx item for a given mpx account.
+ *
+ * @package Drupal\media_mpx\Form
+ */
+class UpdateMediaItemForAccount extends FormBase {
+
+  /**
+   * The mpx Accounts storage.
+   *
+   * @var \Drupal\Core\Entity\EntityStorageInterface
+   */
+  private $accountsStorage;
+
+  /**
+   * The media storage.
+   *
+   * @var \Drupal\media\MediaStorage
+   */
+  private $mediaStorage;
+
+  /**
+   * The update video service.
+   *
+   * @var \Drupal\media_mpx\Service\UpdateVideoItem\UpdateVideoItem
+   */
+  private $updateVideoItemService;
+
+  /**
+   * UpdateMediaItemForAccount constructor.
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager, UpdateVideoItem $updateVideoItem) {
+    $this->mediaStorage = $entityTypeManager->getStorage('media');
+    $this->accountsStorage = $entityTypeManager->getStorage('media_mpx_account');
+    $this->updateVideoItemService = $updateVideoItem;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('media_mpx.service.update_video_item')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $accounts = $this->accountsStorage->loadMultiple();
+    $account_opts = [];
+
+    foreach ($accounts as $account) {
+      $account_opts[$account->id()] = $account->label();
+    }
+
+    $form['account'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Accounts'),
+      '#description' => $this->t('Choose the account that owns the video you want to update.'),
+      '#options' => $account_opts,
+      '#required' => TRUE,
+    ];
+    $form['guid'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('guid'),
+      '#placeholder' => 'Type the GUID of the mpx item you want to import.',
+      '#required' => TRUE,
+    ];
+
+    $form['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Update item'),
+    ];
+
+    return $form;
+  }
+
+  /**
+   * Submit handler for the 'media_mpx_asset_sync_single_by_account_guid' form.
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $account = $form_state->getValue('account');
+    $guid = $form_state->getValue('guid');
+
+    if (!$media_items = $this->loadMediaEntitiesWithMatchingGuid($guid)) {
+      $this->messenger()->addError($this->t('There are no video items with the entered GUID.'));
```

Comment:
What if I'm importing a new video?

Actually, that makes me think I've suggested the wrong approach here - instead of selecting an account, it should be a media bundle?

Reply:
Right, totally. I actually wondered why this was for updating, and not for both creation and updates. I'll rework this! 🚀

Reply:
Right, 2 hours of mpx pain later: this can't be done. Details in ticket, but essentially retrieving object data by **guid** is not working for me. So we'll have to restrict this UI to **update only** ops, and defer to the upcoming _global id ui_ for both create/update imports.